### PR TITLE
[AU4] Project toolbar. Implement Trim audio outside selection command

### PR DESCRIPTION
**Pull request by Eism**
Resolves: #7240

### CODE REVIEW

**Review by grliszas14 — APPROVED** [review]

File: au4/src/projectscene/view/clipsview/au3/au3wavepainter.cpp [hunk 1]
```diff
@@ -521,7 +521,11 @@ void DrawMinMaxRMS(int channelIndex, QPainter& painter,
         ColorFromQColor(style.clippedPen))
     .SetEnvelope(clip.GetEnvelope());
 
-    waveformPainter.Draw(channelIndex, painter, paintParameters, metrics);
```
Comment (grliszas14) [inline au4/src/projectscene/view/clipsview/au3/au3wavepainter.cpp:524]: actually... why we're dropping metrics passed here before?

File: au4/src/trackedit/internal/au3/au3interaction.h [hunk 2]
```diff
@@ -38,6 +38,8 @@ class Au3Interaction : public ITrackeditInteraction
     bool changeClipStartTime(const trackedit::ClipKey& clipKey, double newStartTime, bool completed) override;
     muse::async::Channel<trackedit::ClipKey, double /*newStartTime*/, bool /*completed*/> clipStartTimeChanged() const override;
 
+    bool trimTrackData(trackedit::TrackId trackId, double begin, double end) override;
```
Comment (igorkorsukov) [inline au4/src/trackedit/internal/au3/au3interaction.h:41]: I think better to use secs_t here

File: au4/src/trackedit/internal/au3/au3trackeditproject.cpp [hunk 3]
```diff
@@ -136,6 +136,12 @@ void Au3TrackeditProject::onTrackAdded(const Track &track)
     notifier.itemAdded(track);
 }
 
+void Au3TrackeditProject::onTrackChanged(const Track &track)
+{
+    async::ChangedNotifier<Track>& notifier = m_tracksChanged;
+    notifier.itemChanged(track);
```
Comment (igorkorsukov) [inline au4/src/trackedit/internal/au3/au3trackeditproject.cpp:142]: hmm, why not just ?
```
m_tracksChanged.itemChanged(track);
```

File: au4/src/projectscene/view/clipsview/au3/au3wavepainter.cpp (hunk 1 above)
Comment (Eism) [reply au4/src/projectscene/view/clipsview/au3/au3wavepainter.cpp:524]: We adjust taking into account the trim left (just above I take the current metrics and adjust fromTime/toTime)

**Pull request merged**